engine/etcdng/etcd.go

```diff
@@ -729,6 +733,13 @@ func (n *ng) deleteKey(key string) error {
 	return convertErr(err)
 }
 
+func (n *ng) periodicallySyncCluster() {
+	for {
```

Comment:
this goroutine should exit when `Close` has been called. I suggest adding close channel and doing

```
select {
     case time.After <-...
      case <- n.closeC <-...
```